src/ripple/app/tx/applySteps.h

```diff
@@ -38,6 +39,134 @@ isTecClaimHardFail(TER ter, ApplyFlags flags)
     return isTecClaim(ter) && !(flags & tapRETRY);
 }
 
+/** Class describing the consequences to the account
+    of applying a transaction if the transaction consumes
+    the maximum XRP allowed.
+*/
+class TxConsequences
+{
+public:
+    /// Describes how the transaction affects subsequent
+    /// transactions
+    enum Category
+    {
+        /// Moves currency around, creates offers, etc.
+        normal = 0,
+        /// Affects the ability of subsequent transactions
+        /// to claim a fee. Eg. `SetRegularKey`
+        blocker
+    };
+
+private:
+    /// Describes how the transaction affects subsequent
+    /// transactions
+    Category category_;
```

Comment:
Making the member variable `Category category_` instead of `bool isBlocker_` feels like an extra level of indirection. It's easy to see what the `bool` variable is all about. The `Category` variable requires looking at the `Category` enum. But it's no problem if you prefer the `Category` version. It's a minor detail one way or the other.

(I agree the enum should be kept for the constructor argument)